### PR TITLE
762 create endpoints for v2index

### PR DESCRIPTION
**Pull request by yangcao77**
**Please specify the area for this PR**
index registry server

**What does does this PR do / why we need it**:
this PR includes the change to new endpoints serve for v2index, and new endpoint looking for devfile of a specific stack version. 
Also includes the filter for `minSchemaVersion` and `maxSchemaVersion` when looking for v2index and also when looking for devfile of `latest` stack version. 

**Which issue(s) this PR fixes**:

Fixes https://github.com/devfile/api/issues/762

**PR acceptance criteria**:

- [x] Test (WIP) 

- [ ] Documentation (WIP)

**How to test changes / Special notes to the reviewer**:
run ./build_registry.sh and deploy the built devfile-index image to cluster.
send GET request to server to test endpoints

### CODE REVIEW

**Review by johnmcollier — CHANGES_REQUESTED** [review]
Generally looks good, just one question.

Could you also add some new integration tests for the new endpoint?

File: index/server/pkg/server/endpoint.go [hunk 1]
```diff
@@ -85,6 +100,41 @@ func serveDevfile(c *gin.Context) {
...
 	}
+
+	// minSchemaVersion and maxSchemaVersion will only be applied if looking for latest stack version
+	if version == "latest" {
+		minSchemaVersion := c.Query("minSchemaVersion")
+		maxSchemaVersion := c.Query("maxSchemaVersion")
+		// check if schema version filters are in valid format.
+		// should only include major and minor version. e.g. 2.1
+		if minSchemaVersion != "" {
+			matched, err := regexp.MatchString(`^([2-9])\.([0-9]+)$`, minSchemaVersion)
```
[the remark this answers is not in the document]
Comment (yangcao77) [reply index/server/pkg/server/endpoint.go:111]: we have discussed to restricted the `minSchemaVersion` and `maxSchemaVersion` version to only check for major & minor version, right? so in this case , for example`2.1.1` falls into `maxSchemaVersion= 2.1`, and `2.2.3` can be filtered using using `minSchemaVersion=2.1`

**Comment by yangcao77** [comment]
@johnmcollier Can I add integration tests in the same PR? I saw the integration tests only tests on production/staging registry, but my changes have not been on staging yet.

**Comment by johnmcollier** [comment]
@yangcao77 it _should_ be fine. The integration tests for PRs are configured to point to the test registry.

And once the PR is merged, this will get deployed in staging automatically.

**Comment by yangcao77** [comment]
@johnmcollier  integration tests added

**Review by johnmcollier — APPROVED** [review]

**Pull request merged**